common/setopt.c

```diff
@@ -175,3 +175,214 @@ GetOptionAndValue(
     }
     goto cleanup;
 }
+
+static
+uint32_t
+_TDNFGetCmdOpt(
+    PTDNF_CMD_ARGS pArgs,
+    const char *pszOptName,
+    PTDNF_CMD_OPT *ppOpt
+    )
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_OPT pOpt = NULL;
+    int nHasOpt = 0;
+
+    for (pOpt = pArgs->pSetOpt;
+         pOpt;
+         pOpt = pOpt->pNext)
+    {
+        if (pOpt->nType == CMDOPT_KEYVALUE &&
+            !strcmp(pOpt->pszOptName, pszOptName))
+        {
+            nHasOpt = 1;
+            break;
+        }
+    }
+
+    if (!nHasOpt)
+    {
+        dwError = ERROR_TDNF_FILE_NOT_FOUND;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    *ppOpt = pOpt;
+error:
+    return dwError;
+}
+
+uint32_t
+TDNFHasOpt(
+    PTDNF_CMD_ARGS pArgs,
+    const char *pszOptName,
+    int *pnHasOpt
+    )
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_OPT pOpt = NULL;
+    int nHasOpt = 0;
+
+    if(!pArgs || !pArgs->pSetOpt ||
+       IsNullOrEmptyString(pszOptName) || !pnHasOpt)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = _TDNFGetCmdOpt(pArgs, pszOptName, &pOpt);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    *pnHasOpt = nHasOpt;
+
+cleanup:
+    return dwError;
+
+error:
+    if (dwError == ERROR_TDNF_FILE_NOT_FOUND)
+    {
+        dwError = 0;
+    }
+    goto cleanup;
+}
+
+uint32_t
+TDNFGetCmdOptValue(
+    PTDNF_CMD_ARGS pArgs,
+    const char *pszOptName,
+    char **ppszOptValue
+    )
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_OPT pOpt = NULL;
+    char *pszOptValue = NULL;
+
+    if(!pArgs || !pArgs->pSetOpt ||
+       IsNullOrEmptyString(pszOptName) || !ppszOptValue)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = _TDNFGetCmdOpt(pArgs, pszOptName, &pOpt);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    if (pOpt->pszOptValue)
+    {
+        dwError = TDNFAllocateString(pOpt->pszOptValue, &pszOptValue);
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    *ppszOptValue = pszOptValue;
+
+cleanup:
+    return dwError;
+
+error:
+    TDNF_SAFE_FREE_MEMORY(pszOptValue);
+    goto cleanup;
+}
+
+/*
+ * if get fails, use the default value (if supplied)
+*/
+uint32_t
+TDNFGetOptWithDefault(
+    PTDNF_CMD_ARGS pArgs,
+    const char *pszOptName,
+    const char *pszDefault,
+    char **ppszOptValue
+    )
+{
+    uint32_t dwError = 0;
+    char *pszOptValue = NULL;
+
+    dwError = TDNFGetCmdOptValue(pArgs, pszOptName, &pszOptValue);
+    if (!IsNullOrEmptyString(pszDefault) &&
+        dwError == ERROR_TDNF_FILE_NOT_FOUND)
+    {
+        dwError = TDNFAllocateString(pszDefault, &pszOptValue);
+    }
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    *ppszOptValue = pszOptValue;
+error:
+    return dwError;
+}
+
+uint32_t
+TDNFSetOptNoOverwrite(
```

Comment:
Is this function used anywhere? Is this added for future use?

Reply:
unused. removed.